cts-java/src/test/java/org/ga4gh/cts/api/references/ReferenceBasesPagingIT.java

```diff
@@ -0,0 +1,152 @@
+        return resp.getSequence();
+    }
+
+    /**
+     * Check that we can request a sequence of bases from
+     * {@link org.ga4gh.ctk.transport.protocols.Client.References#getReferenceBases(String,
+     * ListReferenceBasesRequest)}
+     * using a size equal to the size of the full sequence, and receive the full sequence.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkRequestingRightSizedBaseSequence() throws AvroRemoteException {
+
+        final String fullSequence = getFullSequence();
+        checkChunkOfBases(TestData.REFERENCE_BRCA1_LENGTH, fullSequence);
+    }
+
+    /**
+     * Check that if we request a sequence of bases from
+     * {@link org.ga4gh.ctk.transport.protocols.Client.References#getReferenceBases(String,
+     * ListReferenceBasesRequest)}
+     * using a size larger than the full sequence, it fails with an exception with HTTP status
+     * "Requested Range Not Satisfiable."
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @SuppressWarnings("ThrowableResultOfMethodCallIgnored")
+    @Test
+    public void checkRequestingTooLargeBaseSequenceFails() throws AvroRemoteException {
+
+        final String fullSequence = getFullSequence();
+
+        // try it with a range that's too large by 100%
+        GAWrapperException g =
+                Utils.catchGAWrapperException(() -> checkChunkOfBases(TestData.REFERENCE_BRCA1_LENGTH * 2,
+                                                                      fullSequence));
+        assertThat(g.getHttpStatusCode()).isEqualTo(Utils.HTTP_REQUESTED_RANGE_NOT_SATISFIABLE);
+
+        // try it with a range that's too large by one
+        g = Utils.catchGAWrapperException(() -> checkChunkOfBases(TestData.REFERENCE_BRCA1_LENGTH + 1,
+                                                                  fullSequence));
+        assertThat(g.getHttpStatusCode()).isEqualTo(Utils.HTTP_REQUESTED_RANGE_NOT_SATISFIABLE);
+    }
+
+    /**
+     * Check that we receive expected results when we request a single
+     * chunk of bases from zero to <tt>chunkSize</tt> from
+     * {@link org.ga4gh.ctk.transport.protocols.Client.References#getReferenceBases(String,
+     * ListReferenceBasesRequest)}
+     * using <tt>chunkSize</tt> as the sequence length.
+     *
+     * @param chunkSize        the sequence length we'll request
+     * @param expectedSequence the full base sequence we expect
+     * @throws AvroRemoteException if there's a communication problem or server exception
+     */
+    private void checkChunkOfBases(long chunkSize,
+                                   String expectedSequence)
+            throws AvroRemoteException {
+
+        final String refId = Utils.getValidReferenceId(client);
```

Comment:
The test has a hidden dependency on there being exactly one reference, which I consider to be an issue now - not only later, once more references are added. Ideally this would be fixed before the test is submitted, but if not there should at least be a TODO here (I wouldn't block my +1 on it being fixed here).